activesupport/lib/active_support/values/time_zone.rb

```diff
@@ -248,7 +249,18 @@ def [](arg)
       # A convenience method for returning a collection of TimeZone objects
       # for time zones in the USA.
       def us_zones
-        @us_zones ||= all.find_all { |z| z.name =~ /US|Arizona|Indiana|Hawaii|Alaska/ }
+        country_zones(:us)
+      end
+
+      # A convenience method for returning a collection of TimeZone objects
+      # for time zones in the country specified by its ISO 3166-1 Alpha2 code.
+      def country_zones(country_code)
+        code = country_code.to_s.upcase
+        @country_zones[code] ||=
+          TZInfo::Country.get(code).zone_identifiers.map do |tz_id|
```

Comment:
@arthurnn: I think @Envek is using `@country_zones` (`ThreadSafe::Cache`) correctly here.

PS:
> I've initialized cache instance earlier and only adding key/value pairs to it, never overwriting.

@Envek just wanted to clarify, it is thread-safe to do something like this:
```ruby
cache = ThreadSafe::Cache.new
cache[:foo] = [1,2]
cache[:foo] = [1,2,3] # rewriting/updating cache is ok
cache[:foo] += [4] # this is just syntactic sugar for cache[:foo] = cache[:foo] + [4]
```

however this wouldn't be thread-safe:
```ruby
cache = ThreadSafe::Cache.new
cache[:foo] = [1,2]
cache[:foo] << 3 # updating ruby Arrays is not thread-safe
```

PS n. 2: `ThreadSafe::Cache` has been merged into https://github.com/ruby-concurrency/concurrent-ruby, so on rebase you'll need to replace `ThreadSafe::Cache.new` with `Concurrent::Map.new`.